clang/include/clang/CIR/Dialect/IR/CIRAttrs.td

```diff
@@ -335,6 +335,35 @@ def CmpThreeWayInfoAttr : CIR_Attr<"CmpThreeWayInfo", "cmp3way_info"> {
   let genVerifyDecl = 1;
 }
 
+//===----------------------------------------------------------------------===//
+// DataMemberPtrLayoutAttr
+//===----------------------------------------------------------------------===//
+
+def DataMemberPtrLayoutAttr
```

Comment:
I don't really get why you are adding this level of abstraction, seems like everything can be directly queried by data layout, please elaborate.

Reply:
Removed this attribute.